src/Catalyst.Core.Lib/Kernel/Kernel.cs

```diff
@@ -0,0 +1,218 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.IO;
+using System.Reflection;
+using System.Security;
+using Autofac;
+using Autofac.Configuration;
+using AutofacSerilogIntegration;
+using Catalyst.Abstractions.Config;
+using Catalyst.Abstractions.Cryptography;
+using Catalyst.Abstractions.Types;
+using Catalyst.Abstractions.Util;
+using Catalyst.Core.Lib.Config;
+using Catalyst.Core.Lib.Util;
+using Catalyst.Protocol.Network;
+using Microsoft.Extensions.Configuration;
+using Serilog;
+using SharpRepository.Ioc.Autofac;
+using SharpRepository.Repository;
+
+namespace Catalyst.Core.Lib.Kernel
+{
+    public sealed class Kernel : IDisposable
+    {
+        private string _withPersistence;
+        private bool _overwrite;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the field '_overwrite' and declare it as a local variable in the relevant methods.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)